### ISSUE TITLE
Components

### ISSUE THREAD

**Issue by DG4ever** [dsuryd/dotNetify#18]
In the app-template I noticed that you are using some components to reuse html code.
This was the thing I missed until now.

> Do you have some documentation about this or maybe a link where I can read more about it?

_Ok it seems that this are simple knockout components. I am not sure how to load a dotnetify BaseVM as a viewmodel of the component, so I don't need to pass every parameter on it's own. Is this even possible?_

It would be great to be able to create views with associated viewmodels that I can reuse everywhere as I do in wpf.

Here is an example. I want to create a text that translates automatically.
Let's call it dictionary-text. 
I got a ViewModel looks like this:
```
public class DictionaryText : BaseVM
{
	public string TableName { get; set; }
	public string Content { get; set; }
	public string TranslatedText { get; set; }

	public DictionaryText()
	{
		TableName = "";
		Content = "";
		TranslatedText = "";
		MainWindowVM.OnLanguageChanged -= LanguageChanged;
		MainWindowVM.OnLanguageChanged += LanguageChanged;
		LanguageChanged(null, new LanguageEventArgs(MainWindowVM._SelectedLanguage));
	}

	private void LanguageChanged(object source, LanguageEventArgs e)
	{
		TranslatedText = CommonTableReader.GetMessage(Content, TableName, e.GetSelectedLanguage());
		Changed(() => TranslatedText);
		PushUpdates();
	}

	public override void Dispose()
	{
		base.Dispose();
		MainWindowVM.OnLanguageChanged -= LanguageChanged;
	}
}
```

Now I want to create a component which can be used everywhere in HTML like this:

`<dictionary-text data-vm="DictionaryTextVM" params="Content: 'Descr_ExampleID', TableName: 'messages'"/>`

In the ViewModel the TableName Property should now be set to `messages` and the Content property should be `Descr_ExampleID`.

The Text the dictionary-text is going to display is the TranslatedText Property.

Is there any way I can achieve something like this?

**Comment by DG4ever** [comment]
Thanks, I already noticed that I can't bind to a Viewmodel inside another Viewmodel. That's quite annoying sometimes.
What I also noticed is that the vmItemKey doesn't work when it is used inside a component. Not really a problem, as I solved it by overriding the OnUnresolvedUpdate method.

Unfortionatly I need the DictionaryText very often inside a another Viewmodel. Is there a way to pass initial values to a property as it is done with data-vm-args for a viewmodel?
